src/IO/ParallelReadBuffer.h

```diff
@@ -134,7 +72,7 @@ class ParallelReadBuffer : public SeekableReadBufferWithSize
 
     SynchronizedArenaWithFreeLists arena;
```

Comment:
This should be deleted.
Arena goes directly to the global allocator if the blocks are too large (larger than 63KB) which is true for HTTPReader. I.e. the arena always went to the global allocator and ParallelReadBuffer had just unnecessary mutex contention.